### ISSUE TITLE
MA0124: Does not report invalid type for log parameter when using Serilog

### ISSUE THREAD

**Issue by jason-simard-sagewright** [meziantou/Meziantou.Analyzer#643]
### Version of the Meziantou.Analyzer NuGet package

2.0.103

### Rule Identifier

MA0124

### Target Framework

NET 6.0

### C# Language version

C# 10

### Description

After adding the Meziantou.Analyzer and Meziantou.Analyzer.Annotations nuget packages to a .NET 6 API, the analyzer will only correctly flag invalid types for a log parameter when using the default Microsoft.Extensions.Logging.ILogger.  

If you use the exact same configuration but swap out the default ILogger for Serilog.ILogger, you get no such warning/error in the editor or at build time. 

I've tried this configuration in both Visual Studio 2022 17.7.6 and Rider 2023.2.3.

I've tried using both the text file and the assembly level attributes.  Both returned the same result with only the Microsoft.Extensions.Logging.ILogger successfully flagging the invalid parameter types in the editor or during the build with a  warning or error based on the .editorconfig setting.

Is there a way to get the analyzer to flag the invalid parameter types when using Serilog?

### Reproduction Steps

Configuration in the LoggerParameterTypes.txt file

FirstParameter;System.DateTime
SecondParameter;System.String

Added to the .csproj file

    <ItemGroup>
        <AdditionalFiles Include="$(MSBuildThisFileDirectory)\LoggerParameterTypes.txt" />
    </ItemGroup>

OR

Using the assembly attributes

```c#
using Meziantou.Analyzer.Annotations;

[assembly: StructuredLogField("FirstParameter", typeof(DateTime))]
[assembly: StructuredLogField("SecondParameter", typeof(string))]
```

**Comment by jason-simard-sagewright** [comment]
Amazing!  I can confirm it now works perfectly with Serilog in both Visual Studio 2022 and Rider 2023.2.3.
